engine/tests/phpunit/bootstrap.php

```diff
@@ -1,72 +1,45 @@
 <?php
 
-use Zend\Mail\Transport\InMemory as InMemoryTransport;
-
-require_once __DIR__ . '/../../../vendor/autoload.php';
+/**
+ * Instead of relying on this bootstrap, which leads to unreliable global state,
+ * test cases should extend \Elgg\TestCase, which resets service providers
+ * during test initialization making sure altered state does not
+ * flow over to the next test case.
+ * 
+ * @deprecated 2.3
+ */
 
-date_default_timezone_set('America/Los_Angeles');
+if (!defined('PHPUNIT_ELGG_TESTING_APPLICATION')) {
+	// this value is set by phpunit.xml
+	return;
+}
 
-error_reporting(E_ALL | E_STRICT);
+\Elgg\TestCase::bootstrap();
 
 /**
  * Get/set an Application for testing purposes
  *
  * @param \Elgg\Application $app Elgg Application
  * @return \Elgg\Application
+ * @deprecated 2.3 Use elgg() instead
  */
 function _elgg_testing_application(\Elgg\Application $app = null) {
```

Comment:
We can toss them. I thought if we are keeping the bootstrap, we might as well keep the functions for the time being.